src/api/services/FavoriteItemService.ts

```diff
@@ -0,0 +1,136 @@
+import * as Bookshelf from 'bookshelf';
+import { inject, named } from 'inversify';
+import { Logger as LoggerType } from '../../core/Logger';
+import { Types, Core, Targets } from '../../constants';
+import { validate, request } from '../../core/api/Validate';
+import { NotFoundException } from '../exceptions/NotFoundException';
+import { FavoriteItemRepository } from '../repositories/FavoriteItemRepository';
+import { FavoriteItem } from '../models/FavoriteItem';
+import { FavoriteItemCreateRequest } from '../requests/FavoriteItemCreateRequest';
+import { FavoriteItemUpdateRequest } from '../requests/FavoriteItemUpdateRequest';
+import { RpcRequest } from '../requests/RpcRequest';
+
+
+export class FavoriteItemService {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Repository) @named(Targets.Repository.FavoriteItemRepository) public favoriteItemRepo: FavoriteItemRepository,
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+    @validate()
+    public async rpcFindAll( @request(RpcRequest) data: any): Promise<Bookshelf.Collection<FavoriteItem>> {
+        return this.findAll();
+    }
+
+    public async findAll(): Promise<Bookshelf.Collection<FavoriteItem>> {
+        return this.favoriteItemRepo.findAll();
+    }
+
+    @validate()
+    public async rpcFindOne( @request(RpcRequest) data: any): Promise<FavoriteItem> {
+        return this.findOne(data.params[0]);
+    }
+
+    public async findOne(id: number, withRelated: boolean = true): Promise<FavoriteItem> {
+        const favoriteItem = await this.favoriteItemRepo.findOne(id, withRelated);
+        if (favoriteItem === null) {
+            this.log.warn(`FavoriteItem with the id=${id} was not found!`);
+            throw new NotFoundException(id);
+        }
+        return favoriteItem;
+    }
+
+    public async findByItemAndProfile(data: any): Promise<FavoriteItem> {
+        return this.favoriteItemRepo.findByItemAndProfile(data);
```

Comment:
if you're passing an object containing search params, then create a FavoriteSearchParams object to validate the params. see ListingItemService.search for example. also when we're passing a search param object, rename the method to search to match the naming convention we have.